packages/eslint-plugin/tests/rules/no-extra-parens.test.ts

```diff
@@ -242,6 +242,11 @@ f<(number | string)[]>(['a', 1])
         },
       },
     }),
+    {
+      code: `
+const a = (['a', 'b'] satisfies A[]).includes('a');
```

Comment:
[Testing] Can we also have a few cases showing the rule still complaining when it should?

* `((['a', 'b'] satisfies A[]))`
* `((['a', 'b']) satisfies A[])`
* `(['a', 'b'] satisfies (A[]))`

...and also, what if the parenthesis are unnecessary? We'll want at least one test case:

* `const value = (['a', 'b'] satisfies A[]);`

Reply:
...you know, I don't remember what my point was on that last one 😅

Reply:
Marking as resolved 😂